packages/server/src/users.ts

```diff
@@ -14,6 +15,17 @@ export interface PersistentUser extends UserResponse {
   username: string;
   password_hash: string;
   login_type: "persistent";
+  rating?: number;
+}
+
+export async function updateUserRating(
+  user_id: string,
+  new_rating: number,
+): Promise<UpdateResult<GuestUser | PersistentUser>> {
```

Comment:
I could be missing something, but according to [mongodb source](https://github.com/mongodb/node-mongodb-native/blob/00be68f/src/operations/update.ts#L33), `UpdateResult` is not generic - I'm not sure what specifying a template parameter does.